test/kongintegration/containers/kong.go

```diff
@@ -0,0 +1,83 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/test/internal/testenv"
+)
+
+const (
+	kongAdminPort       = "8001"
+	kongProxyPort       = "8000"
+	defaultRouterFlavor = "expressions"
+)
+
+// Kong represents a docker container running Kong.
+type Kong struct {
+	container testcontainers.Container
+}
+
+// NewKong spawns a docker container running Kong (its image is determined by environment variables).
+// It sets up a cleanup function that will terminate the container when the test finishes.
+func NewKong(ctx context.Context, t *testing.T) Kong {
```

Comment:
nit: I rather prefer moving it above type `Kong` to have method IP(..) closer to the struct